### PR TITLE
Skip dependency evaluation with `--skip-dependencies`

### LINKED ISSUE #3783: `--skip-dependencies` does not bypass dependency correlation
### Brief description of your issue

When `--skip-dependencies` is provided, I expect that the client should behave as if dependencies are not present in the manifest and install the requested package without its dependencies.

### Steps to reproduce

I'm gonna take the example of what's causing the issue on my machine, but this can be generally applied to any error that's related to dependencies in the manifest. (e.g., MinVersion specified in the manifest does not exist in the repo anymore and it's causing issues in the install flow)

Due to #3696, `Microsoft.WindowsAppRuntime.1.3` is not correctly correlated on my machine because of which I cannot install devhome properly.

1. `winget install devhome`
2. See -
image
3. `winget install devhome --skip-dependencies`

### Expected behavior

Dependency correlation and installation should be skipped, and the base package should be installed

### Actual behavior

image

### Environment

```shell
Windows Package Manager v1.6.2771
Copyright (c) Microsoft Corporation. All rights reserved.

Windows: Windows.Desktop v10.0.22621.2428
System Architecture: X64
Package: Microsoft.DesktopAppInstaller v1.21.2771.0

Winget Directories
-----------------------------------------------------------------------------------------------------------------------
Logs                               %LOCALAPPDATA%\Packages\Microsoft.DesktopAppInstaller_8wekyb3d8bbwe\LocalState\Diag…
User Settings                      %LOCALAPPDATA%\Packages\Microsoft.DesktopAppInstaller_8wekyb3d8bbwe\LocalState\sett…
Portable Links Directory (User)    %LOCALAPPDATA%\Microsoft\WinGet\Links
Portable Links Directory (Machine) C:\Program Files\WinGet\Links
Portable Package Root (User)       %LOCALAPPDATA%\Microsoft\WinGet\Packages
Portable Package Root              C:\Program Files\WinGet\Packages
Portable Package Root (x86)        C:\Program Files (x86)\WinGet\Packages
Installer Downloads                %USERPROFILE%\Downloads

Links
---------------------------------------------------------------------------
Privacy Statement   https://aka.ms/winget-privacy
License Agreement   https://aka.ms/winget-license
Third Party Notices https://aka.ms/winget-3rdPartyNotice
Homepage            https://aka.ms/winget
Windows Store Terms https://www.microsoft.com/en-us/storedocs/terms-of-sale

Admin Setting                             State
--------------------------------------------------
LocalManifestFiles                        Enabled
BypassCertificatePinningForMicrosoftStore Disabled
InstallerHashOverride                     Enabled
LocalArchiveMalwareScanOverride           Disabled
```

### PR DESCRIPTION
**Pull request by mdanish-kh**
- [x] I have signed the [Contributor License Agreement](https://cla.opensource.microsoft.com/microsoft/winget-pkgs).
- [x] This pull request is related to an issue.
	- Resolves #3783

-----

 ###### Microsoft Reviewers: [Open in CodeFlow](https://microsoft.github.io/open-pr/?codeflow=https://github.com/microsoft/winget-cli/pull/3784)

### CODE REVIEW

File: src/AppInstallerCLICore/Workflows/DependenciesFlow.cpp [hunk 1]
```diff
@@ -249,9 +249,9 @@ namespace AppInstaller::CLI::Workflow
         const auto& rootInstaller = context.Get<Execution::Data::Installer>();
         const auto& rootDependencies = rootInstaller->Dependencies;
 
-        if (rootDependencies.Empty())
+        if (rootDependencies.Empty() || context.Args.Contains(Execution::Args::Type::SkipDependencies))
```
Comment (yao-msft) [inline src/AppInstallerCLICore/Workflows/DependenciesFlow.cpp:252]: In theory there should only be one entry for dependencies in the workflow. But I reviewed dependency related workflows and they are scattered, I think we need a refactor to fix the issue. For now, I'm ok adding this block here. But can you move it to the top of this method?

**Comment by yao-msft** [comment]
InstallerWithDependencies_SkipDependencies test failed because there's an unused override (mock).

By the way, the SkipDependencies arg check should check for user settings like other places as well.

Please let me know if any more info is needed. Thanks.

**Comment by mdanish-kh** [comment]
> Please let me know if any more info is needed. Thanks.

Thank you. I added the user setting check in the latest commit.

Can I know how can I locally run/debug the tests defined in `WorkFlowCommon.cpp`? I notice there is a script `Run-TestsInPackage.ps1` that sets up the test environment and calls `AppInstallerCLITests.exe`. I could get those tests to run but not sure how the flow is setup for `WorkFlowCommon.cpp` which is causing the build issue here.

Looking at the code, it seems to me that `OverrideOpenDependencySource` would be getting unused after this change but I can't verify this locally

**Comment by yao-msft** [comment]
All unit tests are compiled into AppInstallerTests.exe. They are located under `src\<arch>\<debug|release>\AppInstallerCliTests`

A quick way to run the above test is `AppInstallerTests.exe InstallerWithDependencies_SkipDependencies`.

**Comment by mdanish-kh** [comment]
> All unit tests are compiled into AppInstallerTests.exe. They are located under `src\<arch>\<debug|release>\AppInstallerCliTests`
> 
> A quick way to run the above test is `AppInstallerTests.exe InstallerWithDependencies_SkipDependencies`.

Thanks! Turns out I did get the test to pass but since it wasn't failing I couldn't see a reference to `WorkflowCommon.cpp` and unused override error🤦 

Looks like `InstallerWithDependencies_IgnoreDependenciesSetting` test case also requires the same fix

**Review by yao-msft — APPROVED** [review]

**Pull request merged**